From 7982963fc42060e20e5b9907acbda487c8a96462 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kamil=20Go=C5=82da?= <kamil.golda@protonmail.com>
Date: Fri, 21 Feb 2025 14:18:59 +0100
Subject: [PATCH 12/20] clk: qcom: dispcc-sm6125: Do not use parking clk_ops
 for MDSS
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The clock parking at registration introduced in commit 01a0a6cc8cfd
("clk: qcom: Park shared RCGs upon registration") messes up the clocks
when applied to the disp_cc_mdss_mdp_clk_src clock. In result the
display is completely black during boot, so it's needed to opt out of
the parking.

Fixes: 01a0a6cc8cfd ("clk: qcom: Park shared RCGs upon registration")
Signed-off-by: Kamil Gołda <kamil.golda@protonmail.com>
---
 drivers/clk/qcom/dispcc-sm6125.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/qcom/dispcc-sm6125.c b/drivers/clk/qcom/dispcc-sm6125.c
index 27a736657..3c01ea69b 100644
--- a/drivers/clk/qcom/dispcc-sm6125.c
+++ b/drivers/clk/qcom/dispcc-sm6125.c
@@ -269,7 +269,7 @@ static struct clk_rcg2 disp_cc_mdss_mdp_clk_src = {
 		.name = "disp_cc_mdss_mdp_clk_src",
 		.parent_data = disp_cc_parent_data_3,
 		.num_parents = ARRAY_SIZE(disp_cc_parent_data_3),
-		.ops = &clk_rcg2_shared_ops,
+		.ops = &clk_rcg2_shared_no_init_park_ops,
 	},
 };
 
-- 
2.56.0

